control-admin/src/main/java/fi/nls/oskari/control/admin/GetAllRolesAndPermissionTypesHandler.java

```diff
@@ -0,0 +1,121 @@
+package fi.nls.oskari.control.admin;
+
+import static fi.nls.oskari.control.ActionConstants.KEY_ID;
+import static fi.nls.oskari.control.ActionConstants.KEY_NAME;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.oskari.permissions.PermissionService;
+import org.oskari.permissions.model.PermissionType;
+
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.control.ActionParamsException;
+import fi.nls.oskari.control.RestActionHandler;
+import fi.nls.oskari.domain.Role;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.service.ServiceException;
+import fi.nls.oskari.service.UserService;
+import fi.nls.oskari.util.JSONHelper;
+import fi.nls.oskari.util.ResponseHelper;
+
+@OskariActionRoute("GetAllRolesAndPermissionTypes")
+public class GetAllRolesAndPermissionTypesHandler extends RestActionHandler {
+    
+    private Logger log = LogFactory.getLogger(GetAllRolesAndPermissionTypesHandler.class);
+    private UserService userService = null;
+    private PermissionService permissionsService = null;
+
+    private final static String JSKEY_ID = "id";
+    private final static String JSKEY_NAME = "name";
+    private final static String JSKEY_ROLES = "roles";
+    private final static String JSKEY_PERMISSION_TYPES = "permissionTypes";
+
+    @Override
+    public void init() {
+        try {
+            userService = UserService.getInstance();
+        } catch (Exception ex) {
+            log.error(ex, "Unable to initialize User service!");
+        }
+        try {
+            permissionsService = OskariComponentManager.getComponentOfType(PermissionService.class);
+        } catch (Exception ex) {
+            log.error(ex, "Unable to initialize permission service!");
+        }
+    }
+
+    @Override
+    public void preProcess(ActionParameters params) throws ActionException {
+        params.requireAdminUser();
+
+        if (userService == null) {
+            throw new ActionParamsException("User service not initialized");
+        }
+        if (permissionsService == null) {
+            throw new ActionParamsException("PermissionsService service not initialized");
+        }
+    }
+
+    @Override
+    public void handleGet(ActionParameters params) throws ActionException {
+        try {
+            
+            final JSONObject root = new JSONObject();
+            getRoles(root);
+            getPermissionTypes(root,params.getLocale().getLanguage());
```

Comment:
Refactor getRoles() and getPermissionTypes() to return the array and construct the response object here since methods don't really need the root as param.

Reply:
Implemented approach was attempt to get less code in handleGet method by putting role and permission related queries and loops to their own private methods. I now changed the implementation such way that root json is now constucted in handleGet method. Role and permission related json arrays are still constructed in private methods.